gateway/src/constructors/express/routes/index.js

```diff
@@ -1,30 +1,45 @@
-const path = require('path')
-const cors = require('cors')
-const router = require('express').Router()
+const path = require("path");
+const cors = require("cors");
+const router = require("express").Router();
 
 module.exports = (context = {}) => {
-  const APIRouter = require('./api')(context)
+ const { sequelize, models } = context;
+ const APIRouter = require("./api")(context);
+ const KEYRouter = require("./key")(context);
 
-  router.use('/api', (req, res, next) => {
-    console.log('TODO - authenticate...')
-    next()
-  })
-
-  if (process.env.NODE_ENV === 'development') {
-    router.use('/api', cors(), APIRouter)
-  } else {
-    router.use('/api', APIRouter)
+ router.use("/api", async (req, res, next) => {
+  console.log("TODO - authenticate....");
+  try {
+   const { apiKey = "*" } = req.query;
+   const Op = sequelize.Op;
+   const result = await models.APIKey.findAll({ where: { api_key: apiKey } });
+   if (result.length == 0) {
+    return res.status(401).json({ error: "API KEY NOT VALID" });
+   }
+   next();
+  } catch (e) {
+   console.error(e);
+   return res.status(401).json({ error: [e] });
   }
+ });
+
+ router.use("/model/api/key", KEYRouter)
```

Comment:
Something like: `/public/v1/apikeys` will work. It's standard practice with REST routes to have version namespaces and to use plural path tokens.